extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaHeaderReader.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.kafkainput;
+
+import org.apache.druid.data.input.InputRow;
+import java.io.IOException;
+
+public interface KafkaHeaderReader
+{
+  InputRow read() throws IOException;
```

Comment:
I'm not sure all the complexity of the semantics of InputRow is warranted to read headers. InputRow has dimensions, metrics, and timestamps, which might make it confusing to understand what should be returned.
Maybe we can keep this interface simple and just return a map or key/value pairs?

Reply:
I started writing header reader first and initially was under the assumption that header reader will be an independent format reader and hence had fixed it to inputRow, later when we blended it with one final format I did not change it. This is an oversight, will definitely change this, thanks Xavier.

Reply:
done